go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -0,0 +1,1130 @@
+				existingConfig.InsertOverwriteFailed(*newConfig)
+				return fmt.Errorf("unable to ensure iptables rules: %v", err)
+			}
+		} else {
+			if err := c.iptablesManager.EnsureRules(utiliptables.TableNAT, iptChainName, utiliptables.ProtocolIPv4, []iptables.RuleArg{newConfig.ipTableRule}); err != nil {
+				existingConfig.InsertOverwriteFailed(*newConfig)
+				return fmt.Errorf("failed to ensure rules (%+v) in chain %s: %v", newConfig.ipTableRule, iptChainName, err)
+			}
+		}
+		existingConfig.insertOverwrite(*newConfig)
+	}
+	return nil
+}
+
+func (c *Controller) getAllEIPs() ([]*eipv1.EgressIP, error) {
+	eips, err := c.eIPLister.List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("failed to list EgressIPs: %v", err)
+	}
+	return eips, nil
+}
+
+// RepairNode generates whats expected and what is seen on the node and removes any stale configuration. This should be
+// called at Controller startup.
+func (c *Controller) RepairNode() error {
+	links, err := util.GetNetLinkOps().LinkList()
+	if err != nil {
+		return fmt.Errorf("failed to ensure IP is correctly configured becase we could not list links: %v", err)
+	}
+	// get address map for each interface -> addresses/mask
+	// also map address/mask -> interface name
+	assignedAddr := sets.New[string]()
+	assignedAddrStrToAddrs := make(map[string]netlink.Addr)
+	assignedIPRoutes := sets.New[string]()
+	assignedIPRouteStrToRoutes := make(map[string]routemanager.RoutesPerLink)
+	assignedIPRules := sets.New[string]()
+	assignedIPRulesStrToRules := make(map[string]netlink.Rule)
+	assignedIPTableV4Rules := sets.New[string]()
+	assignedIPTableV6Rules := sets.New[string]()
+	assignedIPTablesV4StrToRules := make(map[string]iptables.RuleArg)
+	assignedIPTablesV6StrToRules := make(map[string]iptables.RuleArg)
+
+	for _, link := range links {
+		link := link
+		linkName := link.Attrs().Name
+		linkIdx := link.Attrs().Index
+		addresses, err := linkmanager.GetExternallyAvailableAddresses(link, c.v4, c.v6)
+		if err != nil {
+			return fmt.Errorf("unable to get link addresses for link %s: %v", linkName, err)
+		}
+		for _, address := range addresses {
+			if address.Label == linkmanager.GetAssignedAddressLabel(linkName) {
+				addressStr := address.String()
+				assignedAddr.Insert(addressStr)
+				assignedAddrStrToAddrs[addressStr] = address
+			}
+		}
+		filter, mask := filterRouteByLinkTable(linkIdx, getRouteTableID(linkIdx))
+		existingRoutes, err := util.GetNetLinkOps().RouteListFiltered(netlink.FAMILY_ALL, filter, mask)
+		if err != nil {
+			return fmt.Errorf("unable to get route list using filter (%s): %v", filter.String(), err)
+		}
+		for _, existingRoute := range existingRoutes {
+			route := routemanager.ConvertNetlinkRouteToRoute(existingRoute)
+			routeStr := route.String()
+			assignedIPRoutes.Insert(routeStr)
+			assignedIPRouteStrToRoutes[routeStr] = routemanager.RoutesPerLink{Link: link,
+				Routes: []routemanager.Route{route}}
+		}
```

Comment:
Why do we care about routes of links that do not have the label set?

Reply:
Should Ignore this edge case to reduce risk of cleaning up routes that we dont own? I am assuming every route table we create, we own and it will be 1000 + link index.

Reply:
Hmm if that is the case why we would base that on existing links instead of just removing all the routes above 1000 that we don't recognize?
I guess it is just safer to stuck to labeled links.